src/java/htsjdk/variant/variantcontext/filter/FilteringIterator.java

```diff
@@ -0,0 +1,91 @@
+package htsjdk.variant.variantcontext.filter;
+
+import htsjdk.samtools.util.CloseableIterator;
+import htsjdk.samtools.util.CloserUtil;
+import htsjdk.samtools.util.PeekableIterator;
+import htsjdk.variant.variantcontext.VariantContext;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * A simple filtering iterator for VariantContexts that takes a base iterator
+ * and a VariantContextFilter. Similiar in idea to the SAMRecordFilter. Different from
+ * Picard's VariantFilter in that it actually filters the
+ */
+public class FilteringIterator implements CloseableIterator<VariantContext> {
```

Comment:
Hmm. that library isn't available in htsjdk. I hesitate to add it. What do others think?